flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/rules/physical/stream/StreamExecTemporalUdtfJoinRule.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.plan.rules.physical.stream
+
+import org.apache.flink.table.plan.`trait`.FlinkRelDistribution
+import org.apache.flink.table.plan.nodes.FlinkConventions
+import org.apache.flink.table.plan.nodes.logical._
+import org.apache.flink.table.plan.nodes.physical.stream.StreamExecTemporalTableJoin
+import org.apache.flink.table.plan.util.TemporalJoinUtil.containsTemporalJoinCondition
+import org.apache.flink.table.plan.util.{FlinkRelOptUtil, WindowJoinUtil}
+
+import org.apache.calcite.plan.{RelOptRule, RelOptRuleCall, RelTraitSet}
+import org.apache.calcite.plan.RelOptRule.{any, operand}
+import org.apache.calcite.rel.RelNode
+import org.apache.calcite.rel.core.JoinRelType
+
+import java.util
+
+class StreamExecTemporalUdtfJoinRule
+  extends RelOptRule(
+    operand(
+      classOf[FlinkLogicalJoin],
```

Comment:
I am really confused about `LogicalCorrelateToTemporalTableJoinRule`.. It seems that really TemporalTableJoin keeps `FlinkLogicalJoin` until `StreamExecTemporalUdtfJoinRule`?

Reply:
Both of lookup join and temporal tablefunction join are correlates before optimization. We need to convert them to joins in the first step. 
I renamed them to `LogicalCorrelateToJoinFromTemporalTableRule` and `LogicalCorrelateToJoinFromTemporalTableFunctionRule`, what do you think?

Reply:
`LogicalCorrelateToLookupJoinRule` and `LogicalCorrelateToTemporalJoinRule`?

Reply:
I don't think `LogicalCorrelateToLookupJoinRule` is a good idea. Because it doesn't translate Correlate to a Lookup Join here, but only convert correlate to a join. Currently the join will be converted to lookup join in physical. However, we may also support convert the join to temporal join in the future.

Reply:
OK, LGTM for `LogicalCorrelateToJoinFromTemporalTableRule`, you can add some comments, because it need be distinguished to flink-planner.

Reply:
Sure